src/hotspot/share/runtime/crac.cpp

```diff
@@ -405,6 +406,7 @@ Handle crac::checkpoint(jarray fd_arr, jobjectArray obj_arr, bool dry_run, jlong
   Universe::heap()->set_cleanup_unused(true);
   Universe::heap()->collect(GCCause::_full_gc_alot);
   Universe::heap()->set_cleanup_unused(false);
+  G1UncommitRegionTask::wait_if_active();
```

Comment:
Should we create a generic virtual method on heap (noop by default) to make this part of code GC-agnostic?

Reply:
Yes, that makes sense, done, thanks.